### PR TITLE
#549 No whitespace on form submission

### LINKED ISSUE #549: [Enhancement] - Some forms/modals need beginning and trailing whitespace deleted before running the function
### Describe the Current Behavior/Feature:
Noticed this issue on 2 modals/forms:
**1. Add Contacts**
**2. New Message**

**ADD CONTACTS:**
If user has a whitespace before or after text in `First Name`, `Last Name`, or `Username`, it does not error but leaves the whitespace in the data - not great.

**NEW MESSAGE:**
If user has a whitespace before the text or after the text in the `To:` field, it will error.

---

### Proposed Behavior/Feature:
Need to have these inputs validated, and have beginning and trailing whitespace deleted prior to running functions involved.

---

### Rationale:
Not super noticeable to user why failing (due to whitespace)

---

### Additional context:
https://github.com/codeforpdx/PASS/assets/71395931/a96ee9c9-0a26-4bf7-b132-f525301c094d

### PR DESCRIPTION
**Pull request by milofultz**
Resolves #549
 
**1.** Strips leading/trailing whitespace before submission
**2.** Cleans up some other cruft I found along the way

### CODE REVIEW

File: src/components/Modals/AddContactModal.jsx [hunk 1]
```diff
@@ -85,14 +86,16 @@ const AddContactModal = ({ addContact, showAddContactModal, setShowAddContactMod
...
   return (
     <Dialog
       open={showAddContactModal}
       aria-labelledby="dialog-title"
       onClose={() => setShowAddContactModal(false)}
     >
-      <FormSection title="Add Contact">
-        <form onSubmit={handleAddContact} autoComplete="off">
+      <FormSection title={formTitle}>
+        <form aria-label={formTitle} onSubmit={handleAddContact} autoComplete="off">
```
Comment (xscottxbrownx) [inline src/components/Modals/AddContactModal.jsx]: I believe `aria-label` should be used when there is no content in the DOM that can be used to label this element.

So, here I believe we should use `aria-labelledby` and reference the `<FormSection>` element which labels this form via it's title.

**Review by timbot1789 — APPROVED** [review]

**Review by leekahung — APPROVED** [review]
Think I'm good with this too.

File: src/components/Modals/AddContactModal.jsx (hunk 1 above)
Comment (milofultz) [reply src/components/Modals/AddContactModal.jsx]: Fixed, thank you!

File: src/components/Modals/NewMessageModal.jsx [hunk 2]
```diff
@@ -129,103 +138,106 @@ const NewMessageModal = ({ showModal, setShowModal, oldMessage = '', toField = '
           minHeight: '90%'
         }}
       >
-        <form onSubmit={(e) => handleSubmit(e)} autoComplete="off">
```
Comment (milofultz) [inline src/components/Modals/NewMessageModal.jsx:132]: This looks like a lot, but I just wrapped the form in a FormSection and removed the typography below.

**Comment by milofultz** [comment]
I added a FormSection to the NewMessage modal because I realized that should probably be done the same way as the other one, and now they both have labels.

File: src/components/Modals/NewMessageModal.jsx [hunk 3]
```diff
@@ -89,7 +90,14 @@ const NewMessageModal = ({ showModal, setShowModal, oldMessage = '', toField = '
       addNotification('error', 'Please enter a message');
```
Comment (xscottxbrownx) [inline src/components/Modals/NewMessageModal.jsx]: This wasn't from you, but I don't think line 84-89 get used anymore. I think there is validation on each input now so leaving these blank are not an option.

File: src/components/Modals/NewMessageModal.jsx [hunk 4]
```diff
@@ -129,103 +138,106 @@ const NewMessageModal = ({ showModal, setShowModal, oldMessage = '', toField = '
...
+              type="text"
+              name="title"
+              id="title"
+              onChange={(e) => handleChange(e)}
+              required
+              label="Subject"
+              inputProps={{ maxLength: '48' }}
+              fullWidth
+            />
+            <TextField
```
Comment (xscottxbrownx) [inline src/components/Modals/NewMessageModal.jsx]: This text field should be conditionally rendered. Even on a New Message modal, it still shows in the DOM currently.  Maybe `oldMessage && <TextField>`?

File: src/components/Modals/NewMessageModal.jsx [hunk 5]
```diff
@@ -129,103 +138,106 @@ const NewMessageModal = ({ showModal, setShowModal, oldMessage = '', toField = '
...
-              >
-                Cancel
-              </Button>
-              <Button
-                variant="contained"
-                type="submit"
-                color="primary"
-                startIcon={<CheckIcon />}
-                fullWidth
+              InputLabelProps={{
```
Comment (xscottxbrownx) [inline src/components/Modals/NewMessageModal.jsx]: I'm not sure this label shrink is needed.  I didn't notice any overlap from label and input value on any device I tested.

File: src/components/Form/FormSection.jsx [hunk 6]
```diff
@@ -14,11 +14,12 @@ import { useTheme } from '@mui/material/styles';
  * @name FormSection
  * @param {object} Props - A React prop that consists of
  * that consist of title and children (see {@link formSectionProps})
+ * @param {string} Props.id - The `id` attribute of the element containing the `title`
  * @param {string} Props.title - Title of form section
  * @param {React.ReactElement} Props.children - JSX Element of the wrapped form
  * @returns {React.JSX.Element} - The FormSection Component
  */
-const FormSection = ({ title, children }) => {
+const FormSection = ({ title, id, children }) => {
```
Comment (leekahung) [inline src/components/Form/FormSection.jsx]: Just a nitpick, but think we could use something like `htmlId` for the prop name as to not get confused for where the `id` attribute is attributed to (the component itself or an element within the component).

**Review by xscottxbrownx — CHANGES_REQUESTED** [review]
See my previous comments, but the big one for me is to make that one conditionally rendered so it's not adding needlessly to the DOM.

**Comment by milofultz** [comment]
> See my previous comments, but the big one for me is to make that one conditionally rendered so it's not adding needlessly to the DOM.

Got all these fixes in! If you're happy, feel free to merge it in.

File: src/components/Modals/AddContactModal.jsx [hunk 7]
```diff
@@ -91,8 +92,8 @@ const AddContactModal = ({ addContact, showAddContactModal, setShowAddContactMod
       aria-labelledby="dialog-title"
       onClose={() => setShowAddContactModal(false)}
     >
-      <FormSection title="Add Contact">
-        <form onSubmit={handleAddContact} autoComplete="off">
+      <FormSection title="Add Contact" id="add-contact-form">
```
Comment (leekahung) [inline src/components/Modals/AddContactModal.jsx]: With the update to using `headingId` for the prop for `<FormSection>` in place of `id` which we'll reserve for elements within the form component, we can update this to `headingId="add-contact-form"`

File: src/components/Modals/NewMessageModal.jsx [hunk 8]
```diff
@@ -129,103 +130,105 @@ const NewMessageModal = ({ showModal, setShowModal, oldMessage = '', toField = '
...
-                fullWidth
-              >
-                Cancel
-              </Button>
-              <Button
-                variant="contained"
-                type="submit"
-                color="primary"
-                startIcon={<CheckIcon />}
+        <FormSection title={oldMessage ? 'Reply To' : 'New Message'} id="new-message-form">
```
Comment (leekahung) [inline src/components/Modals/NewMessageModal.jsx]: Think we can update this to use `headingId="new-message-form"`

File: src/components/Modals/AddContactModal.jsx (hunk 7 above)
Comment (milofultz) [reply src/components/Modals/AddContactModal.jsx]: gah, brain off right now, thank you

**Comment by leekahung** [comment]
Looks like there was a pretty significant divergence that happened, but I got everything patched up. 👍

It should be good to go from my end @milofultz. Think you've address the points I've mentioned. Awaiting other reviewers.

**Review by xscottxbrownx — APPROVED** [review]

**Pull request merged**